tests/test_serialization.py

```diff
@@ -266,6 +280,25 @@ def test_decimal_field_special_values_not_permitted(self, user):
         assert isinstance(m7s, decimal.Decimal)
         assert m7s.is_zero() and m7s.is_signed()
 
+    def test_decimal_field_fixed_point_representation(self, user):
+        """
+        Test we get fixed-point string representation for a Decimal number that would normally
+        output in engineering notation.
+        """
+        user.m1 = '0.00000000100000000'
+
+        field = fields.Decimal()
```

Comment:
How about caching serialization result instead of calling serialization twice?
```python
result = fields.Decimal().serialize('m1', user)
assert isinstance(result, decimal.Decimal)
assert result == decimal.Decimal(user.m1)

result = fields.Decimal(as_string=True).serialize('m1', user)
assert isinstance(result, basestring)
assert result == user.m1
```